a10_nlbaas2oct/driver.py

```diff
@@ -78,26 +80,45 @@
                 help='Name of LBaaS service provider'),
     cfg.BoolOpt('ignore_l7rule_status', default=False,
                 required=True,
-                help='Migrate all pending create l7rules')
+                help='Migrate all pending create l7rules'),
+    cfg.StrOpt('default_flavor_id', required=False,
+               help='Default flavor-id that will be used for each loadbalancer in the run'),
+    cfg.ListOpt('flavor_id_list',
+                help='List of flavor id\'s to migrate by mapping with lb_id_list')
 ]
 
 cfg.CONF.register_cli_opts(cli_opts)
 cfg.CONF.register_opts(migration_opts, group='migration')
 
 
-def _migrate_flavor(LOG, a10_config, o_session, device_name):
+def _migrate_flavor(LOG, a10_config, o_session, device_name, conf_flavor_id_list):
     # Translate the name expressions into an Octavia flavor
     LOG.info('Migrating name expressions to flavors')
     fl_id = None
+    fl_id_list = []
     try:
-        flavor_data = nexpr2fl.create_flavor_data(a10_config, device_name)
-        if flavor_data:
-            fp_id = nexpr2fl.create_flavorprofile(o_session, flavor_data)
-            fl_id = nexpr2fl.create_flavor(o_session, fp_id)
+        if conf_flavor_id_list:
+            for conf_flavor_id in conf_flavor_id_list:
+                fl_id = db_utils.get_flavor_id(o_session, conf_flavor_id)
```

Comment:
I believe we should make it a strict requirement that the flavor can be retrieved. If it cannot, we should skip over the lb in the migration and tell them that the specified flavor could not be found.

Otherwise the user could get into a situation where they need to delete lots of lb's that were mistakenly migrated before a flavor was created.